Added current_frame_image property to the Sprite class.

DoAltPlusF4 opened the pull request:
This is a small utility property that users may find helpful when creating visual effects or just general animation.
Just thought I'd make this PR just before 1.5.8 is released.

benmoran56 commented:
Not to get off on a tangent here, but this is some code we were tossing around in the past: 

```py
class AnimationPlayer(pyglet.sprite.Sprite):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._paused = False

    def pause_animation(self):
        if self._paused or not hasattr(self, '_animation'):
            return
        pyglet.clock.unschedule(self._animate)
        self._paused = True

    def resume_animation(self):
        if not self._paused or not hasattr(self, '_animation'):
            return
        frame = self._animation.frames[self._frame_index]
        self._texture = frame.image.get_texture()
        self._next_dt = frame.duration
        if self._next_dt:
            pyglet.clock.schedule_once(self._animate, self._next_dt)
        self._paused = False

    @property
    def frame_index(self):
        return self._frame_index

    @frame_index.setter
    def frame_index(self, index):
        # Bound to available number of frames
        self._frame_index = max(0, min(index, len(self._animation.frames)-1))
```

Maybe it's a good idea to add these to the Sprite class finally.

DoAltPlusF4 commented:
I replaced the current_frame_image property with those new functions and properties.
I have also moved the _frame_index attribute to the top of the class, with things like `_animation`, `_batch`, `_paused` and `_rotation`:
```py
class Sprite(event.EventDispatcher):
    # ...

    _batch = None
    _animation = None
    _frame_index = 0
    _paused = False
    _rotation = 0
    _opacity = 255
    _rgb = (255, 255, 255)
    _scale = 1.0
    _scale_x = 1.0
    _scale_y = 1.0
    _visible = True
    _vertex_list = None

    # ...
